app/feature/feature-addon-purchase/src/main/kotlin/com/hedvig/android/feature/addon/purchase/ui/customize/CustomizeTravelAddonViewModel.kt

```diff
@@ -0,0 +1,127 @@
+  @Composable
+  override fun MoleculePresenterScope<CustomizeTravelAddonEvent>.present(
+    lastState: CustomizeTravelAddonState,
+  ): CustomizeTravelAddonState {
+    var currentState by remember { mutableStateOf(lastState) }
+    var loadIteration by remember { mutableIntStateOf(0) }
+    var selectedOptionInDialog by remember { mutableStateOf<TravelAddonQuote?>(null) }
+    CollectEvents { event ->
+      when (event) {
+        CustomizeTravelAddonEvent.Reload -> loadIteration++
+        is CustomizeTravelAddonEvent.ChooseOptionInDialog -> {
+          selectedOptionInDialog = event.option
+        }
+
+        CustomizeTravelAddonEvent.ChooseSelectedOption -> {
+          val state = currentState as? CustomizeTravelAddonState.Success ?: return@CollectEvents
+          val optionInDialog = selectedOptionInDialog ?: return@CollectEvents
+          currentState = state.copy(
+            currentlyChosenOption = optionInDialog,
+          )
+        }
+
+        CustomizeTravelAddonEvent.SetOptionBackToPreviouslyChosen -> {
+          val state = currentState as? CustomizeTravelAddonState.Success ?: return@CollectEvents
+          selectedOptionInDialog = state.currentlyChosenOption
+        }
+
+        CustomizeTravelAddonEvent.ClearNavigation -> {
+          val state = currentState as? CustomizeTravelAddonState.Success ?: return@CollectEvents
+          currentState = state.copy(quoteToNavigateFurther = null)
+        }
+
+        CustomizeTravelAddonEvent.SubmitSelected -> {
+          val state = currentState as? CustomizeTravelAddonState.Success ?: return@CollectEvents
+          currentState = state.copy(
+            quoteToNavigateFurther = state.currentlyChosenOption,
+          )
+        }
+      }
+    }
+
+    LaunchedEffect(loadIteration) {
+      currentState = CustomizeTravelAddonState.Loading
+      getTravelAddonOfferUseCase.invoke(insuranceId).fold(
+        ifLeft = { error ->
+          currentState = CustomizeTravelAddonState.Failure(error.message)
+        },
+        ifRight = { offer ->
+          selectedOptionInDialog = offer.addonOptions[0]
+          currentState = CustomizeTravelAddonState.Success(
+            travelAddonOffer = offer,
+            currentlyChosenOption = offer.addonOptions[0],
+            currentlyChosenOptionInDialog = selectedOptionInDialog,
+          )
+        },
+      )
+    }
+
+    val state = currentState
+    return when (state) {
+      is CustomizeTravelAddonState.Failure, CustomizeTravelAddonState.Loading -> state
+      is CustomizeTravelAddonState.Success -> state.copy(
+        currentlyChosenOptionInDialog = selectedOptionInDialog,
+      )
+    }
+  }
+}
+
+internal sealed interface CustomizeTravelAddonState {
+  data object Loading : CustomizeTravelAddonState
+
+  data class Success(
+    val travelAddonOffer: TravelAddonOffer,
+    val currentlyChosenOption: TravelAddonQuote,
+    val currentlyChosenOptionInDialog: TravelAddonQuote?,
```

Comment:
The bug there was just me using the wrong property yesterday, so no big thing there:) I think I did it like you're suggesting the first time with tiers, but I did not like how it turned out, don't remember exactly, but it sometimes behaved weird (was one recomposition late? or maybe smth else). This may be more reliable in the end - though more code as well. Wdyt?